### PR TITLE
Con 2367 input fields validation bug p2sprint10

### PR DESCRIPTION
**Pull request by karpagamNallasamy**
Bug #3261,#3256,3288,#3270,#3262. buddy testing done by Nishadi

### CODE REVIEW

File: api/CcsSso.Shared/Domain/Helpers/UtilityHelper.cs [hunk 1]
```diff
@@ -74,12 +74,20 @@ public static bool IsPasswordValidForRequiredCharactors(string password)
 
     public static bool IsUserNameValid(string Name)
     {
+      bool IsRestrictedChar = false;
       try
       {
+        string[] tokens = "+ * % # ! $".Split();
```
Comment (VijayHirudayasamy-bc) [inline api/CcsSso.Shared/Domain/Helpers/UtilityHelper.cs:80]: Can you use normal array with list of item in it string[] cars = {"Volvo", "BMW", "Ford", "Mazda"};

File: api/CcsSso.Shared/Domain/Helpers/UtilityHelper.cs [hunk 2]
```diff
@@ -107,11 +115,21 @@ public static bool IsUserNameLengthValid(string Name)
 
     public static bool IsGroupNameValid(string Name)
     {
+      bool IsRestrictedChar = false;
       try
       {
-        Regex regex = new Regex(RegexExpression.VALID_GROUP_NAME);
-        Match match = regex.Match(Name);
-        return match.Success;
+        string[] tokens = "* % ! $".Split();
```
Comment (VijayHirudayasamy-bc) [inline api/CcsSso.Shared/Domain/Helpers/UtilityHelper.cs:121]: same here

File: api/CcsSso.Shared/Domain/Helpers/UtilityHelper.cs [hunk 3]
```diff
@@ -121,11 +139,21 @@ public static bool IsGroupNameValid(string Name)
 
     public static bool IsContactPointNameValid(string Name)
     {
+      bool IsRestrictedChar = false;
       try
       {
-        Regex regex = new Regex(RegexExpression.VALID_CONTACT_NAME);
-        Match match = regex.Match(Name);
-        return match.Success;
+        string[] tokens = "+ * % # ! $".Split();
```
Comment (VijayHirudayasamy-bc) [inline api/CcsSso.Shared/Domain/Helpers/UtilityHelper.cs:145]: here too

File: api/CcsSso.Core.Service/External/OrganisationGroupService.cs [hunk 4]
```diff
@@ -187,11 +195,22 @@ public async Task UpdateGroupAsync(string ciiOrganisationId, int groupId, Organi
...
 
       //All other special characters not specified in accepted. min 3 max 256
-      if (!UtilityHelper.IsGroupNameValid(organisationGroupRequestInfo.GroupName.Trim()))
+      if (organisationGroupRequestInfo.GroupName != null && !UtilityHelper.IsGroupNameValid(organisationGroupRequestInfo.GroupName.Trim()))
       {
         throw new CcsSsoException(ErrorConstant.ErrorInvalidGroupName);
       }
 
+      if (!string.IsNullOrWhiteSpace(organisationGroupRequestInfo.GroupName))
+      {
```
Comment (VijayHirudayasamy-bc) [inline api/CcsSso.Core.Service/External/OrganisationGroupService.cs:204]: Can be keep all the validation as a one function

**Review by VijayHirudayasamy-bc — COMMENTED** [review]
Please check the comments

**Comment by karpagamNallasamy** [comment]
Can you use normal array with list of item in it string[] cars = {"Volvo", "BMW", "Ford", "Mazda"} – done

each validation different API call. make common function not good. also it takes time now.

I am trying many time to reach you by teams call.



It is very urgent. Nishadi,Tina asking me consciously. please merge it.


Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows

________________________________

**Review by VijayHirudayasamy-bc — APPROVED** [review]

**Pull request merged**